### PR TITLE
TST: Refactor test_maybe_match_name and test_hash_pandas_object

### PR DESCRIPTION
**Pull request by minggli**
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]

File: pandas/tests/util/test_hashing.py [hunk 1]
```diff
@@ -142,39 +142,35 @@ def test_multiindex_objects(self):
...
+        tm.makeTimeSeries(),
+        tm.makeTimedeltaIndex(),
+        tm.makePeriodIndex(),
+        Series(tm.makePeriodIndex()),
+        Series(pd.date_range('20130101', periods=3, tz='US/Eastern')),
+        MultiIndex.from_product([range(5), ['foo', 'bar', 'baz'],
+                                 pd.date_range('20130101', periods=2)]),
+        MultiIndex.from_product([pd.CategoricalIndex(list('aabc')), range(3)])
+    ])
+    def test_hash_pandas_object(self, obj):
```
Comment (WillAyd) [inline pandas/tests/util/test_hashing.py]: Was anything changed here besides indentation?

Comment (minggli) [reply pandas/tests/util/test_hashing.py]: no, just indentation.

**Comment by minggli** [comment]
@WillAyd further comments welcome.

**Comment by WillAyd** [comment]
lgtm - thanks!

**Comment by minggli** [comment]
@gfyoung

**Review by gfyoung — APPROVED** [review]
LGTM!

cc @jreback

**Comment by jreback** [comment]
thanks @minggli ! keep em coming!

**Pull request merged**